python/cuspatial/cuspatial/tests/test_geodataframe.py

```diff
@@ -321,7 +321,7 @@ def test_boolmask(gpdf, df_boolmask):
 
 
 def test_memory_usage(gs):
```

Comment:
We struggle with tests asserting things about memory usage a lot in RAPIDS. This type of test is very flaky and changes with dependencies, platform, etc.

Does this test serve a critical purpose? Can we remove or skip it? cc: @harrism

Reply:
For context, we were getting `224` in Python 3.9 and 3.10, but we get `228` in Python 3.11.

Reply:
It looks like @thomcom added the test in PR: https://github.com/rapidsai/cuspatial/pull/604

Maybe he knows more about this test and how best to address this issue?

Reply:
He is currently on leave.

@bdice How have we handled these issues before? Disabling the tests or fixing them?

Reply:
Let's mark this test as `xfail` (or skip if that doesn't work)

Reply:
```suggestion
@pytest.mark.xfail(reason="Size discrepancies between Python versions. See https://github.com/rapidsai/cuspatial/issues/1352")
def test_memory_usage(gs):
```

Reply:
`xfail` is a good choice here.

For awareness, in cudf we enable `xfail_strict`. This causes tests to fail if an `xfail`'d test _passes_. (Because it was expected to fail, and did not.) That isn't enabled here, so it's safe to `xfail`. For some versions it will report its status as `xpass` (unexpected pass).

https://github.com/rapidsai/cudf/blob/3b228e2c6d3ec39fcba553c63d53a56760dc1ca6/python/cudf/cudf/tests/pytest.ini#L6